File: src/main/java/coupon/CouponStore.java
package coupon;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public interface CouponStore {

    List<Coupon> findIssuedCouponsByMemberId(UUID memberId);

    Map<CouponType, Integer> remainingCountMap();

    List<Coupon> issue(List<Coupon> coupons);
}

File: src/main/kotlin/coupon/CouponStore.kt
package org.example.coupon

import coupon.CouponType
import java.util.UUID

interface CouponStore {

    fun findIssuedCouponsByMemberId(memberId: UUID): List<Coupon>

    fun remainingCountMap(): Map<CouponType, Int>

    fun issue(coupons: List<Coupon>): List<Coupon>
}